lib/Finance/MIFIR/CONCAT.pm

```diff
@@ -58,10 +56,8 @@ sub mifir_concat {
 sub _process_name {
     my ($str) = @_;
     $str = lc($str);
-    $str =~ s/$_/$romanization->{$_}/g for keys %$romanization;
     $str =~ s/$_\s+//g for (@{$config->{titles}}, @{$config->{prefixes}});
-    $str =~ s/’//g;    # our iconv does not handle this correctly, it returns empty string if we have it
-    $str = $converter->convert($str);
+    $str =~ s/$_/$romanization->{$_}/g for keys %$romanization;
```

Comment:
The `%romanization` hash is pretty long and you are compiling and running a separate regex for each element. Why don't you combine them in one regex and compile it once. Something along these lines:

```perl
my $re=join '', keys %$romanization;
$re=qr/([$re])/;            # provided the key in the hash are not any special characters like \, - or ]

sub _process_name {
...
    $str =~ s/$re/$romanization->{$1}/ge;
}
```

Depending on how many titles and prefixes there are that might also be a better approach for line 59.